src/Akka.Hosting.TestKit/TestKit.cs

```diff
@@ -0,0 +1,214 @@
+
+        public ActorRegistry ActorRegistry => Host.Services.GetRequiredService<ActorRegistry>();
+        
+        public TimeSpan StartupTimeout { get; }
+        public string ActorSystemName { get; }
+        public ITestOutputHelper? Output { get; }
+        public LogLevel LogLevel { get; }
+
+        private TaskCompletionSource<Done> _initialized = new TaskCompletionSource<Done>();
+
+        protected TestKit(string? actorSystemName = null, ITestOutputHelper? output = null, TimeSpan? startupTimeout = null, LogLevel logLevel = LogLevel.Information)
+        : base(Assertions)
+        {
+            ActorSystemName = actorSystemName ?? "test";
+            Output = output;
+            LogLevel = logLevel;
+            StartupTimeout = startupTimeout ?? TimeSpan.FromSeconds(10);
+        }
+        
+        protected virtual void ConfigureHostConfiguration(IConfigurationBuilder builder)
+        { }
+        
+        protected virtual void ConfigureAppConfiguration(HostBuilderContext context, IConfigurationBuilder builder)
+        { }
+
+        protected virtual void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+        { }
+        
+        private void InternalConfigureServices(HostBuilderContext context, IServiceCollection services)
+        {
+            ConfigureServices(context, services);
+            
+            services.AddAkka(ActorSystemName, async (builder, provider) =>
+            {
+                builder.AddHocon(DefaultConfig, HoconAddMode.Prepend);
+                if (Config is { })
+                    builder.AddHocon(Config, HoconAddMode.Prepend);
+
+                builder.ConfigureLoggers(logger =>
+                {
+                    logger.LogLevel = ToAkkaLogLevel(LogLevel);
+                    logger.ClearLoggers();
+                    logger.AddLogger<TestEventListener>();
+                });
+
+                if (Output is { })
+                {
+                    builder.StartActors(async (system, registry) =>
+                    {
+                        var extSystem = (ExtendedActorSystem)system;
+                        var logger = extSystem.SystemActorOf(Props.Create(() => new LoggerFactoryLogger()), "log-test");
+                        await logger.Ask<LoggerInitialized>(new InitializeLogger(system.EventStream));
+                    });
+                }
+
+                await ConfigureAkka(builder, provider);
+
+                builder.AddStartup((system, registry) =>
+                {
+                    base.InitializeTest(system, (ActorSystemSetup)null!, null, null);
+                    _initialized.SetResult(Done.Instance);
+                });
+            });
+        }
+
+        protected virtual Config? Config { get; } = null;
+        
+        protected virtual void ConfigureLogging(ILoggingBuilder builder)
+        { }
+
+        protected virtual Task ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
```

Comment:
Let's make this `abstract` - force the user to spell out what they want to use as part of their tests.